### PR TITLE
promote: Allow providing app-interface checkout directory

### PR DESCRIPTION
**Pull request by NautiluX**
* Add argument to pass in app-interface checkout directory
* Use default location if none is passed in
* Don't silently fail if osdctl is called outside of a git repository

Jira: [OSD-20520](https://issues.redhat.com//browse/OSD-20520)

### CODE REVIEW

File: cmd/promote/git/app_interface.go [hunk 1]
```diff
@@ -23,21 +24,49 @@ type Service struct {
 	} `yaml:"resourceTemplates"`
 }
 
-func BootstrapOsdCtlForAppInterfaceAndServicePromotions() {
-	_, err := getBaseDir()
-	if err != nil {
-		log.Fatal(err)
+type AppInteface struct {
```
Comment (AlexVulaj) [inline cmd/promote/git/app_interface.go:27]: small typo

Comment (NautiluX) [reply cmd/promote/git/app_interface.go]: thanks
fixed the type typo

**Pull request merged**